src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.TargetFrameworkInference.targets

```diff
@@ -193,6 +193,24 @@ Copyright (c) .NET Foundation. All rights reserved.
                  FormatArguments="$(MinimumOSPlatform);$(TargetPlatformVersion)"/>
   </Target>
 
+  <Target Name="_CheckForInvalidTargetPlatformVersion"
```

Comment:
How does this interact with the invalid platform error?  If you have an invalid platform and specify a TargetPlatformVersion, then we should only give you the invalid platform error, and not the invalid platform version error.  Can you add a test case covering this?